### DISCUSSION TITLE
How to configure webhook with self-signed certificate?

### DISCUSSION THREAD

**Discussion by mixalezhnev** [bukhalo/nestjs-telegraf#778] [Q&A]
It's straightforward in Telegraf [docs](https://github.com/telegraf/telegraf#webhooks), but not clearly by nestjs-telegraf docs. Anyway, does anyone have an understanding of how to properly configure a webhook connection with a self-signed certificate using this library?

**Comment by evilsprut** [comment]
First, a self-signed certificate cannot be set for a route or for a telegraph (specially). Self-signed certificate can be set for web server. So you have two options, [install a certificate for your NestJs server](https://docs.nestjs.com/faq/multiple-servers) or [create a separate server for the Telegraf webhook](https://telegraf.js.org/#webhooks). 

Example 1:
```typescript
const app = await NestFactory.createApplicationContext(AppModule);
const bot = app.get(getBotToken());

const tlsOptions = {
  key: fs.readFileSync('server-key.pem'),
  cert: fs.readFileSync('server-cert.pem'),
  ca: [
    fs.readFileSync('client-cert.pem')
  ]
}

bot.telegram.setWebhook('https://server.tld:8443/secret-path', {
  certificate: { source: fs.readFileSync('server-cert.pem') }
})

bot.startWebhook('/secret-path', tlsOptions, 8443)
```

Example 2:
```typescript
const httpsOptions = {
  key: fs.readFileSync('./secrets/private-key.pem'),
  cert: fs.readFileSync('./secrets/public-certificate.pem'),
};
const app = await NestFactory.create(AppModule, {
  httpsOptions,
});
const bot = app.get(getBotToken());

app.use(bot.webhookCallback('/secret-path'));
```

**Reply by mixalezhnev** [reply]
Unfortunately, that wouldn't work because `bot.startWebhook` doesn't return middleware

**Reply by evilsprut** [reply]
I need some time to approve that, i reply later

**Reply by mixalezhnev** [reply]
I'll wait, thx

**Reply by evilsprut** [reply]
Checkout updated comment
